tests/rbac_test.go

```diff
@@ -0,0 +1,227 @@
+				output, err := ktests.RunCommand("oc", args...)
+				Expect(err).ToNot(HaveOccurred())
+				return output
+			}, time.Minute*10).Should(Equal("Running"))
+
+			By("Connect via console.")
+			vm, err := virtClient.VirtualMachineInstance(NamespaceTestRandom).Get(vmi.Name, &metav1.GetOptions{})
+			expecter, err := ktests.LoggedInCirrosExpecter(vm)
+			Expect(err).ToNot(HaveOccurred())
+			defer expecter.Close()
+
+			By("Checking the response from VNC server.")
+			response, err := tests.VNCConnection(NamespaceTestRandom, vmi.Name)
+			Expect(response).To(Equal("RFB 003.008"))
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Check VMI should stop.")
+			args = []string{"stop", vmi.Name}
+			_, err = ktests.RunCommand("virtctl", args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Delete a VM with yaml.")
+			args = []string{"delete", "-f", vmi.Manifest}
+			_, err = ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Delete a namespace")
+			args = []string{"delete", "project", NamespaceTestRandom}
+			_, err = ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	DescribeTable("RBAC Authorization", func(role, namespace string, con bool) {
+		By(fmt.Sprintf("Add %s to user %s", role, tests.UsernameTestUser))
+		args := []string{"adm", "policy", "add-role-to-user", role, tests.UsernameTestUser, "-n", namespace}
+		_, err := ktests.RunCommand("oc", args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		if con == false	{
+			createResourcesToTestViewRole()
+		}
+
+		By(fmt.Sprintf("Login with the user %s", tests.UsernameTestUser))
+		args = []string{"login", "-u", tests.UsernameTestUser, "-p", "123456"}
+		_, err = ktests.RunCommand("oc", args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Create a VM with yaml.")
+		args = []string{"create", "-f", vmi.Manifest, "-n", namespace}
+		_, err = ktests.RunCommand("oc", args...)
+		if con == true {
+			Expect(err).ToNot(HaveOccurred())
+		} else {
+			Expect(err).To(HaveOccurred())
+		}
+
+		By("Check VMI should create.")
+		args = []string{"start", vmi.Name, "-n", namespace}
+		_, err = ktests.RunCommand("virtctl", args...)
+		if con == true {
+			Expect(err).ToNot(HaveOccurred())
+		} else {
+			Expect(err).To(HaveOccurred())
+		}
+
+		By("Check the vmi is running.")
+		args = []string{"get", "vmi", vmi.Name, "--template", "{{.status.phase}}", "-n", namespace}
+		result := Eventually(func() string {
+		output, err := ktests.RunCommand("oc", args...)
+		if con == true {
+			Expect(err).ToNot(HaveOccurred())
+		} else {
+			Expect(err).To(HaveOccurred())
+		}
+		return output
+		}, time.Minute*10)
+		if con == true {
+			result.Should(Equal("Running"))
+		}
+
+		By("Connect via console.")
+		vm, err := virtClient.VirtualMachineInstance(namespace).Get(vmi.Name, &metav1.GetOptions{})
+		if con == true {
+			expecter, err := ktests.LoggedInCirrosExpecter(vm)
+			Expect(err).ToNot(HaveOccurred())
+			defer expecter.Close()
```

Comment:
We don't really want to keep the `expecter` open for the whole `It` closure.
```suggestion
			expecter.Close()
```